From patchwork Tue Feb 28 10:36:10 2023
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Lucas Tanure <lucas.tanure@collabora.com>
X-Patchwork-Id: 13154673
Return-Path: 
 <linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@lists.infradead.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from bombadil.infradead.org (bombadil.infradead.org
 [198.137.202.133])
	(using TLSv1.2 with cipher ECDHE-RSA-AES256-GCM-SHA384 (256/256 bits))
	(No client certificate requested)
	by smtp.lore.kernel.org (Postfix) with ESMTPS id 8E4C5C64EC7
	for <linux-rockchip@archiver.kernel.org>;
 Tue, 28 Feb 2023 10:37:11 +0000 (UTC)
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed;
	d=lists.infradead.org; s=bombadil.20210309; h=Sender:
	Content-Transfer-Encoding:Content-Type:List-Subscribe:List-Help:List-Post:
	List-Archive:List-Unsubscribe:List-Id:MIME-Version:References:In-Reply-To:
	Message-Id:Date:Subject:Cc:To:From:Reply-To:Content-ID:Content-Description:
	Resent-Date:Resent-From:Resent-Sender:Resent-To:Resent-Cc:Resent-Message-ID:
	List-Owner; bh=LUcd5WbYJuADhTE1aGSieAvXwz+ikDGOTwxbThAI1EE=; b=eNLTAPmtzNl0hv
	Bn0OJ6j7TJg/kquh93mUthDRaLgU0X1IDW+PQux36F0tdADuWuqIk4DHudfEQBviPxB+k2oOMLnl1
	Mh+I9BeLmWFbr0ER/4XLwDwRIg8a50wQF9lWqzhpDSjsLlrJeoCUz1OmWCff+pmkQhyxnmXOyOGjb
	jvdklccQ4tkOAeHx19kaavYvYgmx5Yznm98HGBtlE5wI2myEsP+PAvJPk2VwBPs0q8wpBsyaa5EkZ
	F3lPnH9M0roeEFz94zPLFZYh5OiQHrf1RJKSpYPEAIBv/NMTiZslGdhzlNfFA6sDdWFcdO2B8y18l
	ZsITi0RozharjJ0vwutQ==;
Received: from localhost ([::1] helo=bombadil.infradead.org)
	by bombadil.infradead.org with esmtp (Exim 4.94.2 #2 (Red Hat Linux))
	id 1pWxLx-00ChXG-N8; Tue, 28 Feb 2023 10:37:01 +0000
Received: from madras.collabora.co.uk ([2a00:1098:0:82:1000:25:2eeb:e5ab])
	by bombadil.infradead.org with esmtps (Exim 4.94.2 #2 (Red Hat Linux))
	id 1pWxLH-00ChDS-NX; Tue, 28 Feb 2023 10:36:21 +0000
Received: from cryzen.lan (cpc87451-finc19-2-0-cust61.4-2.cable.virginm.net
 [82.11.51.62])
	(using TLSv1.3 with cipher TLS_AES_256_GCM_SHA384 (256/256 bits)
	 key-exchange X25519 server-signature RSA-PSS (4096 bits) server-digest
 SHA256)
	(No client certificate requested)
	(Authenticated sender: tanureal)
	by madras.collabora.co.uk (Postfix) with ESMTPSA id 6D8EE66021F9;
	Tue, 28 Feb 2023 10:36:17 +0000 (GMT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/simple; d=collabora.com;
	s=mail; t=1677580577;
	bh=76t+fzjT2/wIqxMWU7rwXik9LpuADrrJG6sBa8kaQD4=;
	h=From:To:Cc:Subject:Date:In-Reply-To:References:From;
	b=XeknhX51WTygsqlSaC4hpkGHlVLSTt+duAmPzf4o7+NhUq76fjf9sHzkGPxv5/IJW
	 +w29JoQYS0IUw226UmZySi2XHAbpO1mOGXQdgUr2D/HUU6IJr/WB8PyDZkemWo+h1C
	 DyJVPVlEn0FyhgXleD3KaR/H4Y+zDgkkIMG4q7rTSejzm3vN9trMoe2YtMCVM/GnSJ
	 mr/oSGmyUQESoFPTOMFl1B/a6iSpudWxMkcHiDbmuJn8sutF/7xQ2U1CvcZy0TaP1U
	 sYKxnfUKSaz81GdQTIWtlPugcPtSDAL/vTwNpTqepBHAF01X9Ev9WNgzONc3NGAwhj
	 wLsx+auVGYrfA==
From: Lucas Tanure <lucas.tanure@collabora.com>
To: lucas.tanure@collabora.com
Cc: devicetree@vger.kernel.org,
	heiko@sntech.de,
	kernel@collabora.com,
	krzysztof.kozlowski+dt@linaro.org,
	linux-arm-kernel@lists.infradead.org,
	linux-kernel@vger.kernel.org,
	linux-mmc@vger.kernel.org,
	linux-rockchip@lists.infradead.org,
	robh+dt@kernel.org,
	shawn.lin@rock-chips.com,
	ulf.hansson@linaro.org
Subject: [PATCH v2] arm64: dts: rockchip: rk3588: Add sdmmc node
Date: Tue, 28 Feb 2023 10:36:10 +0000
Message-Id: <20230228103610.25108-1-lucas.tanure@collabora.com>
X-Mailer: git-send-email 2.39.2
In-Reply-To: <20230213152740.359055-2-lucas.tanure@collabora.com>
References: <20230213152740.359055-2-lucas.tanure@collabora.com>
MIME-Version: 1.0
X-CRM114-Version: 20100106-BlameMichelson ( TRE 0.8.0 (BSD) ) MR-646709E3 
X-CRM114-CacheID: sfid-20230228_023619_924193_567F61CC 
X-CRM114-Status: UNSURE (   8.32  )
X-CRM114-Notice: Please train this message.
X-BeenThere: linux-rockchip@lists.infradead.org
X-Mailman-Version: 2.1.34
Precedence: list
List-Id: Upstream kernel work for Rockchip platforms
 <linux-rockchip.lists.infradead.org>
List-Unsubscribe: <http://lists.infradead.org/mailman/options/linux-rockchip>,
 <mailto:linux-rockchip-request@lists.infradead.org?subject=unsubscribe>
List-Archive: <http://lists.infradead.org/pipermail/linux-rockchip/>
List-Post: <mailto:linux-rockchip@lists.infradead.org>
List-Help: <mailto:linux-rockchip-request@lists.infradead.org?subject=help>
List-Subscribe: <http://lists.infradead.org/mailman/listinfo/linux-rockchip>,
 <mailto:linux-rockchip-request@lists.infradead.org?subject=subscribe>
Sender: "Linux-rockchip" <linux-rockchip-bounces@lists.infradead.org>
Errors-To: 
 linux-rockchip-bounces+linux-rockchip=archiver.kernel.org@lists.infradead.org

Add SD Card node for RK3588s and RK3588.

Co-developed-by: Shawn Lin <shawn.lin@rock-chips.com>
Signed-off-by: Shawn Lin <shawn.lin@rock-chips.com>
Signed-off-by: Lucas Tanure <lucas.tanure@collabora.com>
---
Changes Since v1:
  None. Re-submitting this patch alone as the first patch of this series was
  already accepted.
---
 arch/arm64/boot/dts/rockchip/rk3588s.dtsi | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3588s.dtsi b/arch/arm64/boot/dts/rockchip/rk3588s.dtsi
index 005cde61b4b2..fca8503aed8c 100644
--- a/arch/arm64/boot/dts/rockchip/rk3588s.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3588s.dtsi
@@ -1099,6 +1099,21 @@ gmac1_mtl_tx_setup: tx-queues-config {
 		};
 	};
 
+	sdmmc: mmc@fe2c0000 {
+		compatible = "rockchip,rk3588-dw-mshc", "rockchip,rk3288-dw-mshc";
+		reg = <0x0 0xfe2c0000 0x0 0x4000>;
+		interrupts = <GIC_SPI 203 IRQ_TYPE_LEVEL_HIGH 0>;
+		clocks = <&scmi_clk SCMI_HCLK_SD>, <&scmi_clk SCMI_CCLK_SD>,
+			 <&cru SCLK_SDMMC_DRV>, <&cru SCLK_SDMMC_SAMPLE>;
+		clock-names = "biu", "ciu", "ciu-drive", "ciu-sample";
+		fifo-depth = <0x100>;
+		max-frequency = <200000000>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&sdmmc_clk &sdmmc_cmd &sdmmc_det &sdmmc_bus4>;
+		power-domains = <&power RK3588_PD_SDMMC>;
+		status = "disabled";
+	};
+
 	sdhci: mmc@fe2e0000 {
 		compatible = "rockchip,rk3588-dwcmshc";
 		reg = <0x0 0xfe2e0000 0x0 0x10000>;
